### PR TITLE
created input component

### LINKED ISSUE #5: Create Input Component
image

- make sure you can edit the field and capture the input
- src/components/Input

### PR DESCRIPTION
**Pull request by fardinzam**
Screenshot 2023-10-15 at 11 04 17 PM

### CODE REVIEW

File: src/components/Input.jsx [hunk 1]
```diff
@@ -1,5 +1,23 @@
-const Input = () => {
-  return <div>Input</div>;
+"use client";
```
Comment (shahdivyank) [inline src/components/Input.jsx:1]: remove this, not needed on this file

File: src/components/register/Register.jsx [hunk 2]
```diff
@@ -1,5 +1,24 @@
...
-  return <div>Register</div>;
+  const recordUserInfo = (submitEvent) => {
+    submitEvent.preventDefault();
+    const username = submitEvent.target[0].value;
+    console.log(`Username: ${username}`);
+  };
+
+  return (
+    <div>
+      <form onSubmit={recordUserInfo}>
```
Comment (shahdivyank) [inline src/components/register/Register.jsx]: call it handleSubmit

File: src/components/register/Register.jsx [hunk 3]
```diff
@@ -1,5 +1,24 @@
+"use client";
+import Input from "../Input.jsx";
+
 const Register = () => {
-  return <div>Register</div>;
+  const recordUserInfo = (submitEvent) => {
```
Comment (shahdivyank) [inline src/components/register/Register.jsx]: submitEvent should just be e

File: src/components/register/Register.jsx [hunk 4]
```diff
@@ -1,5 +1,24 @@
...
+    submitEvent.preventDefault();
+    const username = submitEvent.target[0].value;
+    console.log(`Username: ${username}`);
+  };
+
+  return (
+    <div>
+      <form onSubmit={recordUserInfo}>
+        <Input
+          name={"Username"}
```
Comment (shahdivyank) [inline src/components/register/Register.jsx]: no curly braces if its just a regular string

File: src/components/register/Register.jsx [hunk 5]
```diff
@@ -1,5 +1,24 @@
...
+          placeholder={"Enter a username"}
+          classes={"mt-5 w-[50%] ml-5"}
```
Comment (shahdivyank) [inline src/components/register/Register.jsx]: w1/2

File: src/components/register/Register.jsx [hunk 6]
```diff
@@ -1,5 +1,24 @@
...
+  const recordUserInfo = (submitEvent) => {
+    submitEvent.preventDefault();
+    const username = submitEvent.target[0].value;
+    console.log(`Username: ${username}`);
+  };
+
+  return (
+    <div>
+      <form onSubmit={recordUserInfo}>
+        <Input
```
Comment (shahdivyank) [inline src/components/register/Register.jsx]: make sure u r able to record info and stuff, so u need to make the usestates

File: src/components/Input.jsx [hunk 7]
```diff
@@ -1,5 +1,21 @@
-const Input = () => {
-  return <div>Input</div>;
+const Input = ({ value, onChange, name, placeholder, classes }) => {
+  return (
+    <div className={`flex items-center w-1/2 text-white ${classes}`}>
+      <h6 className="text-xs px-1 z-40 bg-code-black flex fixed ml-3 mt-[-33px] select-none">
```
Comment (shahdivyank) [inline src/components/Input.jsx]: no h tags, just do p tag or div

Comment (shahdivyank) [reply src/components/Input.jsx]: mt-33px is crazy, try to make it responsive

File: src/components/Input.jsx [hunk 8]
```diff
@@ -1,5 +1,21 @@
-const Input = () => {
-  return <div>Input</div>;
+const Input = ({ value, onChange, name, placeholder, classes }) => {
+  return (
+    <div className={`flex items-center w-1/2 text-white ${classes}`}>
+      <p className="text-xs px-1 z-40 bg-code-black flex fixed ml-3 mb-[3%] select-none">
```
Comment (shahdivyank) [inline src/components/Input.jsx]: why do u need z-40? what's the purpose of fixed and flex, dont think you need fixed. remove the select-none

File: src/components/Input.jsx [hunk 9]
```diff
@@ -1,5 +1,21 @@
...
+        {name}
+      </p>
+      <div className="w-full">
```
Comment (shahdivyank) [inline src/components/Input.jsx]: this should not be needed, the input component should be able to do it without the div

File: src/components/Input.jsx [hunk 10]
```diff
@@ -1,5 +1,21 @@
...
+        <input
+          name={name}
+          placeholder={placeholder}
+          type="text"
```
Comment (shahdivyank) [inline src/components/Input.jsx]: make this a prop in case we need to do diff kinds of type

File: src/components/register/Register.jsx [hunk 11]
```diff
@@ -1,5 +1,33 @@
...
+
+const USERNAME_FIELD = "Username";
+
 const Register = () => {
-  return <div>Register</div>;
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    console.log(`Form data: ${JSON.stringify(formData)}`);
+  };
+  const [formData, setFormData] = useState({ [USERNAME_FIELD]: "" });
```
Comment (shahdivyank) [inline src/components/register/Register.jsx]: call it just data and setData

File: src/components/register/Register.jsx [hunk 12]
```diff
@@ -1,5 +1,33 @@
...
+
+  return (
+    <div>
+      <form onSubmit={handleSubmit}>
+        <Input
+          name={USERNAME_FIELD}
+          placeholder="Enter a username"
+          classes="mt-5 w-1/2 ml-5"
+          value={formData[USERNAME_FIELD]}
+          onChange={(e) => setFormDataFromChangeEvent(e, USERNAME_FIELD)}
```
Comment (shahdivyank) [inline src/components/register/Register.jsx]: handleChange

File: src/components/Input.jsx (hunk 8 above)
Comment (fardinzam) [reply src/components/Input.jsx]: the fixed stops the label from displacing the input field

File: src/components/Input.jsx [hunk 13]
```diff
@@ -1,5 +1,19 @@
-const Input = () => {
-  return <div>Input</div>;
+const Input = ({ value, onChange, name, placeholder, classes, type }) => {
+  return (
+    <div className={`flex items-center w-1/2 text-white ${classes}`}>
+      <p className="text-xs px-1 bg-code-black absolute ml-3 mb-[3%]">{name}</p>
```
Comment (shahdivyank) [inline src/components/Input.jsx:13]: what is the absolute for?

File: src/components/Input.jsx [hunk 14]
```diff
@@ -1,5 +1,19 @@
...
+      <div>
```
Comment (shahdivyank) [inline src/components/Input.jsx]: div isn't needed

File: src/components/register/Register.jsx [hunk 15]
```diff
@@ -1,5 +1,33 @@
...
+
+  return (
+    <div>
+      <form onSubmit={handleSubmit}>
+        <Input
+          name={USERNAME_FIELD}
+          placeholder="Enter a username"
+          classes="mt-5 w-1/2 ml-5"
+          value={data[USERNAME_FIELD]}
+          onChange={(e) => handleChange(e, USERNAME_FIELD)}
```
Comment (shahdivyank) [inline src/components/register/Register.jsx]: this should be done at the input comp level

File: src/components/Input.jsx (hunk 13 above)
Comment (fardinzam) [reply src/components/Input.jsx:13]: the label displaces the input field unless its there
Screenshot 2023-10-22 at 10 56 52 AM

**Review by shahdivyank — APPROVED** [review]

**Pull request merged**